src/Soil/boundary_conditions.jl

```diff
@@ -270,3 +270,43 @@ function soil_boundary_fluxes(bc::NamedTuple, boundary, model, Y, Δz, p, t)
     return ClimaLSM.boundary_flux(bc.water, boundary, Δz, p, t, params),
     ClimaLSM.boundary_flux(bc.heat, boundary, Δz, p, t, params)
 end
+
+
+function ClimaLSM.∂tendencyBC∂Y(
+    model::RichardsModel,
+    ::MoistureStateBC,
+    boundary::ClimaLSM.TopBoundary,
+    Δz,
+    Y,
+    p,
+    t,
+)
+    (; ν, vg_α, vg_n, vg_m, S_s, θ_r) = model.parameters
+    K = ClimaLSM.Domains.top_center_to_surface(p.soil.K)
+    ϑ_l = ClimaLSM.Domains.top_center_to_surface(Y.soil.ϑ_l)
+    return ClimaCore.Fields.FieldVector(;
+        :soil => (;
+            :ϑ_l => @. -K / Δz * dψdθ(ϑ_l, ν, θ_r, vg_α, vg_n, vg_m, S_s) /
+               (2 * Δz)
```

Comment:
Just double checking - should K be negative here? And should the second term be divided by `2\Delta z` or just `\Delta z`? From [eqns 20-22 in the overleaf](https://www.overleaf.com/project/63be02f455f84a77642ef485) for `\partialT \partialtheta`, it looks to me like we don't need the negative sign and the second `\Delta z` term doesn't need to be multiplied by 2. I might be misinterpreting though :)

Reply:
In the overleaf I wrote the BC for the divergence -d/dz(F), but here we are giving the divergence of the tendency..so the minus sign get absorbed (and we get two factors of dz).

I think the multiply by 2 needs to be there because the distance between the first cell center and the top surface is dz_1/2, where dz_1 is the thickness of the layer (face to face).  This is the distance used to estimate the boundary flux. But then the divergence of the flux is (F_bc{1/2}  - F_{3/2})/dz_1.